zun/container/k8s/driver.py

```diff
@@ -850,6 +855,27 @@ def inspect_network(self, network):
     #
 
     def pull_image(self, context, repo, tag, image_pull_policy, image_driver_name, **kwargs):
+        for secret_info in self.get_secrets_for_image(repo, context):
+            # Create a new secret for an existing registry
+            if secret_info["registry"] and not secret_info["secret"]:
+                LOG.info(f"Creating new secret {secret_info['name']}")
+                secret = client.V1Secret()
+                secret.metadata = client.V1ObjectMeta(name=secret_info["name"])
+                secret.type = "kubernetes.io/dockerconfigjson"
+                username = secret_info['registry'].username
+                password = secret_info['registry'].password
+                auth = base64.b64encode(f"{username}:{password}".encode("utf-8")).decode("utf-8")
```

Comment:
Maybe good to pull this out into a little helper

```python
def encode_contents(content_str):
    return base64.b64encode(content_str.encode("utf-8")).decode("utf-8")
```

Actually I think there may be a Zun util for this! Check out `encode_file_data`:

```python
def encode_file_data(data):
    if six.PY3 and isinstance(data, str):
        data = data.encode('utf-8')
    return base64.b64encode(data).decode('utf-8')
```